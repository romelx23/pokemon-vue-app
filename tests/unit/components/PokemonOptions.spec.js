import PokemonOptions from "@/../../src/components/PokemonOption.vue";
import { mount, shallowMount } from "@vue/test-utils";
import { beforeEach, describe, expect } from "vitest";
import { pokemonsMock } from "../mocks/pokemons.mock";

describe("PokemonOptions", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(PokemonOptions, {
      propsData: {
        pokemons: pokemonsMock,
      },
    });
  });

  test("debe de hacer match con el Snapshot", () => {
    // diferencia entre toMatchSnapshot y toMatchInlineSnapshot
    // toMatchSnapshot: el snapshot se guarda en un archivo
    // toMatchInlineSnapshot : el snapshot se guarda en el mismo archivo
    // console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("debe de mostrar las 4 opciones correctamente", () => {
    const li = wrapper.findAll("li");
    expect(li.length).toBe(5);
    li.forEach((element, index) => {
      const pokemon = pokemonsMock[index];
      expect(element.text()).toBe(pokemon.name);
    });
  });
  test('debe emitir "selection" con su respectivo parÃ¡metro al hacer click', () => {
    const [li1, li2, li3, li4, li5] = wrapper.findAll("li");
    // estimulo
    li1.trigger("click");
    li2.trigger("click");
    li3.trigger("click");
    li4.trigger("click");
    li5.trigger("click");
    expect(wrapper.emitted("selectPokemon").length).toBe(5);
    expect(wrapper.emitted("selectPokemon")[0]).toEqual([
      { name: "bulbasaur", id: 1 },
    ]);
    expect(wrapper.emitted("selectPokemon")[1]).toEqual([
      { name: "ivysaur", id: 2 },
    ]);
    // console.log(wrapper.emitted("selectPokemon"));
  });
});
