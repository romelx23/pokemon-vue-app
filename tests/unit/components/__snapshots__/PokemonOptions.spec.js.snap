// Vitest Snapshot v1

exports[`PokemonOptions > debe de hacer match con el Snapshot 1`] = `
"<h3 data-v-5c866bc9=\\"\\">Opciones </h3>
<div class=\\"options-container\\" data-v-5c866bc9=\\"\\">
  <ul data-v-5c866bc9=\\"\\">
    <li data-v-5c866bc9=\\"\\">bulbasaur</li>
    <li data-v-5c866bc9=\\"\\">ivysaur</li>
    <li data-v-5c866bc9=\\"\\">venusaur</li>
    <li data-v-5c866bc9=\\"\\">mew</li>
    <li data-v-5c866bc9=\\"\\">charmeleon</li>
  </ul>
</div>"
`;
